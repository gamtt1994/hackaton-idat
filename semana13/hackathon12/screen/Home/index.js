import React from 'react';
import {View, Text, StyleSheet, SafeAreaView} from 'react-native';
import SocialSlider from '../../components/SocialMedia/ListSlider';
import SongSlider from '../../components/SongSlider/ListSlider';
import Cabecera from '../../components/Cabecera';
import {DataSocialMedia} from '../../components/resources/data/DataSocialMedia';
import {DataSong} from '../../components/resources/data/DataSong';

const Dashboard = ({navigation}) => {
  return (
    <SafeAreaView style={styles.container}>
      <View>
        <View style={styles.header}>
          <Cabecera />
        </View>
        <View style={styles.socialMedia}>
          <Text style={styles.title}>My social Networks</Text>
          <SocialSlider data={DataSocialMedia} />
        </View>
        <View style={styles.podcasts}>
          <Text style={styles.title}>My favorite song</Text>
          <SongSlider
            style={{
              flex: 1,
              marginHorizontal: 10,
              justifyContent: 'center',
              alignItems: 'center',
            }}
            navigation={navigation}
            data={DataSong}
          />
        </View>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginHorizontal: 10,
  },
  title: {
    fontSize: 20,
    marginTop: 10,
    marginHorizontal: 20,
    fontWeight: 'bold',
    marginBottom: 20,
  },
  header: {
    height: 100,
  },
  socialMedia: {
    flex: 1,
    marginHorizontal: 20,
  },
  podcasts: {
    flex: 3,
    marginHorizontal: 20,
  },
});

export default Dashboard;
