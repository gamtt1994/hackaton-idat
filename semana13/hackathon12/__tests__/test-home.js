import React from 'react';
import {shallow} from 'enzyme';
import Home, {styles} from '../screen/Home';

describe('Redering', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Home />);
  });
  it('exists render', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('Testing Text', () => {
    expect(wrapper.find('Text')).toHaveLength(1);
  });

  it('Testing View', () => {
    expect(wrapper.find('View')).toHaveLength(4);
  });

  it('Testing style Text', () => {
    expect(wrapper.find('Text').prop('style')).toEqual(styles.title);
  });

  it('Should match Text snapshot', () => {
    expect(wrapper.find('Text')).toMatchSnapshot();
  });

  it('Should match View snapshot', () => {
    expect(wrapper.find('View')).toMatchSnapshot();
  });
});
