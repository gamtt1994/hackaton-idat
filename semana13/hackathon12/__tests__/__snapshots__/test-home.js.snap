// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redering Should match Text snapshot 1`] = `
<Text
  style={
    Object {
      "fontSize": 20,
      "fontWeight": "bold",
      "marginBottom": 20,
      "marginHorizontal": 20,
      "marginTop": 10,
    }
  }
>
  My social Networks
</Text>
`;

exports[`Redering Should match View snapshot 1`] = `
Array [
  <View>
    <View
      style={
        Object {
          "height": 100,
        }
      }
    >
      <ModalPicker />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
            "marginHorizontal": 20,
            "marginTop": 10,
          }
        }
      >
        My social Networks
      </Text>
      <ListSlider
        data={
          Array [
            Object {
              "id": 1,
              "image": "https://assets.stickpng.com/images/58e91965eb97430e819064f5.png",
              "subtitle": "@facebook",
              "title": "Facebok",
            },
            Object {
              "id": 2,
              "image": "https://cdn.clipart.email/88927c181903aa4cd580e5337a0183dd_twitter-logo-icon-social-media-icon-png-and-vector-twitter-_840-859.png",
              "subtitle": "@Twitter",
              "title": "Twitter",
            },
            Object {
              "id": 3,
              "image": "https://cdn.icon-icons.com/icons2/1121/PNG/512/1486147197-social-media-circled-network04_79460.png",
              "subtitle": "@Youtube",
              "title": "Youtube",
            },
            Object {
              "id": 4,
              "image": "https://geoapps.esri.co/avistamientos/mapita/Ima/inta.png",
              "subtitle": "@Instagram",
              "title": "Instagram",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 3,
          "marginHorizontal": 20,
        }
      }
    >
      <ListSlider
        data={
          Array [
            Object {
              "id": 1,
              "image": "https://pbs.twimg.com/media/EV071nIWoAE_bZX.jpg",
              "subtitle": "Libido",
              "title": "En esta habitación",
            },
            Object {
              "id": 2,
              "image": "https://portal.andina.pe/EDPfotografia2/Thumbnail/2009/08/07/000102504W.jpg",
              "subtitle": "Libido",
              "title": "Ojos de angel",
            },
            Object {
              "id": 3,
              "image": "https://pbs.twimg.com/media/EV071nIWoAE_bZX.jpg",
              "subtitle": "Libido",
              "title": "En tu rostro",
            },
            Object {
              "id": 4,
              "image": "https://portal.andina.pe/EDPfotografia2/Thumbnail/2009/08/07/000102504W.jpg",
              "subtitle": "Libido",
              "title": "Universo",
            },
          ]
        }
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "marginHorizontal": 10,
          }
        }
      />
    </View>
  </View>,
  <View
    style={
      Object {
        "height": 100,
      }
    }
  >
    <ModalPicker />
  </View>,
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
          "marginHorizontal": 20,
          "marginTop": 10,
        }
      }
    >
      My social Networks
    </Text>
    <ListSlider
      data={
        Array [
          Object {
            "id": 1,
            "image": "https://assets.stickpng.com/images/58e91965eb97430e819064f5.png",
            "subtitle": "@facebook",
            "title": "Facebok",
          },
          Object {
            "id": 2,
            "image": "https://cdn.clipart.email/88927c181903aa4cd580e5337a0183dd_twitter-logo-icon-social-media-icon-png-and-vector-twitter-_840-859.png",
            "subtitle": "@Twitter",
            "title": "Twitter",
          },
          Object {
            "id": 3,
            "image": "https://cdn.icon-icons.com/icons2/1121/PNG/512/1486147197-social-media-circled-network04_79460.png",
            "subtitle": "@Youtube",
            "title": "Youtube",
          },
          Object {
            "id": 4,
            "image": "https://geoapps.esri.co/avistamientos/mapita/Ima/inta.png",
            "subtitle": "@Instagram",
            "title": "Instagram",
          },
        ]
      }
    />
  </View>,
  <View
    style={
      Object {
        "flex": 3,
        "marginHorizontal": 20,
      }
    }
  >
    <ListSlider
      data={
        Array [
          Object {
            "id": 1,
            "image": "https://pbs.twimg.com/media/EV071nIWoAE_bZX.jpg",
            "subtitle": "Libido",
            "title": "En esta habitación",
          },
          Object {
            "id": 2,
            "image": "https://portal.andina.pe/EDPfotografia2/Thumbnail/2009/08/07/000102504W.jpg",
            "subtitle": "Libido",
            "title": "Ojos de angel",
          },
          Object {
            "id": 3,
            "image": "https://pbs.twimg.com/media/EV071nIWoAE_bZX.jpg",
            "subtitle": "Libido",
            "title": "En tu rostro",
          },
          Object {
            "id": 4,
            "image": "https://portal.andina.pe/EDPfotografia2/Thumbnail/2009/08/07/000102504W.jpg",
            "subtitle": "Libido",
            "title": "Universo",
          },
        ]
      }
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginHorizontal": 10,
        }
      }
    />
  </View>,
]
`;

exports[`Redering Should match snapsho 1`] = `
<Text
  style={
    Object {
      "fontSize": 20,
      "fontWeight": "bold",
      "marginBottom": 20,
      "marginHorizontal": 20,
      "marginTop": 10,
    }
  }
>
  My social Networks
</Text>
`;
